src/workerd/api/tests/http-socket-test.js

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2017-2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import { connect, convertSocketToFetcher } from 'cloudflare:sockets';
+import { strict as assert } from 'node:assert';
+
+export default {
+  async test(ctrl, env) {
+    // Basic connectivity and GET test
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = convertSocketToFetcher(socket);
+
+      const response = await httpClient.fetch('/ping');
```

Comment:
Why does this accept a path with no hostname? How does this fill in the `Host` header, which is required under HTTP/1.1?

Reply:
I discussed with Kenton and the product does not require no-host on the fetcher -- we should accept a full URL for consistency, such that the host component of the URL is used *only* to provide a value for the `Host` header. The provided `URL` is *not* used for routing.

Reply:
No, Kenton just means that
`httpClient.fetch('/ping')` shouldn't work, instead the following should be required
`httpClient.fetch('https://example.com/ping')`

Reply:
Essentially changing the NO in https://github.com/cloudflare/workerd/pull/4005/files#diff-4e2699e03d914d2336177c99e165ed8e7f912f9bb61af226be284e62c3324f6eR613 to YES